libvast/src/evaluate.cpp

```diff
@@ -143,65 +139,88 @@ struct cell_evaluator<relational_operator::greater_equal> {
 
 template <>
 struct cell_evaluator<relational_operator::in> {
-  static bool evaluate(auto, auto) noexcept {
+  static bool evaluate(auto, const auto&) noexcept {
     return false;
   }
 
-  static bool evaluate(view<std::string> lhs, view<std::string> rhs) noexcept {
+  static bool evaluate(view<std::string> lhs, const std::string& rhs) noexcept {
     return rhs.find(lhs) != view<std::string>::npos;
   }
 
-  static bool evaluate(view<std::string> lhs, view<pattern> rhs) noexcept {
-    return materialize(rhs).search(lhs);
+  static bool evaluate(view<std::string> lhs, const pattern& rhs) noexcept {
+    return rhs.search(lhs);
   }
 
-  static bool evaluate(view<ip> lhs, view<subnet> rhs) noexcept {
+  static bool evaluate(view<ip> lhs, const subnet& rhs) noexcept {
     return rhs.contains(lhs);
   }
 
-  static bool evaluate(view<subnet> lhs, view<subnet> rhs) noexcept {
+  static bool evaluate(view<subnet> lhs, const subnet& rhs) noexcept {
     return rhs.contains(lhs);
   }
 
-  static bool evaluate(auto lhs, view<list> rhs) noexcept {
-    return std::any_of(rhs.begin(), rhs.end(), [lhs](data_view data) {
+  static bool evaluate(auto lhs, const list& rhs) noexcept {
+    return std::any_of(rhs.begin(), rhs.end(), [lhs](const data& element) {
       return caf::visit(
-        [lhs](auto view) noexcept {
+        [lhs](const auto& element) noexcept {
           return cell_evaluator<relational_operator::equal>::evaluate(lhs,
-                                                                      view);
+                                                                      element);
         },
-        data);
+        element);
     });
   }
 };
 
 template <>
 struct cell_evaluator<relational_operator::not_in> {
-  static bool evaluate(auto lhs, auto rhs) noexcept {
+  static bool evaluate(auto lhs, const auto& rhs) noexcept {
     return !cell_evaluator<relational_operator::in>::evaluate(lhs, rhs);
   }
 };
 
 template <>
 struct cell_evaluator<relational_operator::ni> {
-  static bool evaluate(auto lhs, auto rhs) noexcept {
-    return cell_evaluator<relational_operator::in>::evaluate(rhs, lhs);
```

Comment:
> **Note**
> This may not be quite so obvious, but `lhs in rhs` does not translate to `rhs ni lhs` anymore now that `rhs` is a const-reference to a data and `lhs` still a view.

Reply:
Can this be extracted into a separate `re2`-less pull request, just for the 3.0 release?